lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php

```diff
@@ -861,6 +862,21 @@ protected function loadMappingFile($file)
         return $result;
     }
 
+    private function assertValid(string $file): void
+    {
+        $backedUpErrorSetting = libxml_use_internal_errors(true);
+        $document = new \DOMDocument();
+        $document->load($file);
```

Comment:
This will cause duplicated IO, better pass already fetched XML string to this function and use `DOMDocument::loadXML()`.

Reply:
Yeah but it gives worse error messages

Reply:
See https://github.com/doctrine/doctrine2/pull/6728#issuecomment-333406605